File: code_snippets/java/src/main/java/develop/ServingVirtualThreads.java
package develop;

import dev.restate.sdk.HandlerRunner;
import dev.restate.sdk.endpoint.Endpoint;
import java.util.concurrent.Executors;

class ServingVirtualThreads {
  public void virtualThreads(Endpoint.Builder builder) {
    // <start_here>
    builder.bind(
        new Greeter(),
        HandlerRunner.Options.withExecutor(Executors.newVirtualThreadPerTaskExecutor()));
    // <end_here>
  }
}

File: code_snippets/kotlin/src/main/kotlin/develop/ServingVirtualThreads.kt
package develop

import dev.restate.sdk.endpoint.Endpoint
import dev.restate.sdk.kotlin.HandlerRunner
import java.util.concurrent.Executors
import kotlinx.coroutines.asCoroutineDispatcher

class ServingVirtualThreads {
  fun virtualThreads(builder: Endpoint.Builder) {
    // <start_here>
    builder.bind(
        Greeter(),
        HandlerRunner.Options(
            coroutineContext = Executors.newVirtualThreadPerTaskExecutor().asCoroutineDispatcher(),
        ),
    )
    // <end_here>
  }
}
